zappa/zappa.py

```diff
@@ -222,8 +225,17 @@ class Zappa(object):
     boto_session = None
     credentials_arn = None
 
-    def __init__(self, boto_session=None, profile_name=None):
+    def __init__(self, boto_session=None, profile_name=None, aws_region=aws_region):
+        self.aws_region = aws_region
         self.load_credentials(boto_session, profile_name)
+        self.s3_client = self.boto_session.client('s3')
+        self.lambda_client = self.boto_session.client('lambda')
+        self.events_client = self.boto_session.client('events')
+        self.apigateway_client = self.boto_session.client('apigateway')
+        self.logs_client = self.boto_session.client('logs')
+        self.iam_client = self.boto_session.client('iam')
+        self.iam = self.boto_session.resource('iam')
+        self.s3 = self.boto_session.resource('s3')
```

Comment:
I agree it is cleaner - it just needs to not cause problems if they get re-used later on instead of immediately after being opened. I expect it works; just something to think about.